[ci/nav-check] action ignores .wokeignore — fixture phrases in scout.personas.yaml block all PRs

suggested-severity: major

## Observation

The `nav-check` GitHub Action invokes `woke` against the entire repo and flags speciesist phrases that exist in `scout.personas.yaml` (and `AGENTS.md`) **on purpose**, as test fixtures / documentation of the patterns the scanner is supposed to detect. The repo's `.wokeignore` already lists both files, but the action ignores it.

Result: every PR opened against `Open-Paws/no-animal-violence` fails CI on the same set of pre-existing findings, regardless of what the PR actually changes. Currently 4 open PRs blocked: #55, #56, #59, #61.

Sample findings from the failing run:

```
scout.personas.yaml:86:16-45: kill two birds with one stone (error)
scout.personas.yaml:87:16-33: beat a dead horse (error)
scout.personas.yaml:88:16-47: more than one way to skin a cat (error)
scout.personas.yaml:89:16-41: shooting fish in a barrel (error)
scout.personas.yaml:90:16-26: guinea pig (warning)
scout.personas.yaml:91:16-33: flog a dead horse (error)
AGENTS.md:20:44-54: free-range (warning)
Error: files with findings: 2
Process completed with exit code 1
```

## Expected

`nav-check` should not flag intentional fixture content in this repo. Either:

- the action honors the repo's `.wokeignore`, or
- fixtures live under a path that's exempted by other means (path layout, action input flag, abstract markers expanded at runtime, etc.).

PRs that touch unrelated code should pass `nav-check` cleanly.

## Context

**Files involved:**

- `scout.personas.yaml` lines 86–91 — fixture phrases (`kill two birds with one stone`, `beat a dead horse`, `more than one way to skin a cat`, `shooting fish in a barrel`, `guinea pig`, `flog a dead horse`). Confirmed against current main: file is 169 lines, fixture block is in the documented test region.
- `AGENTS.md` line 20 — `free-range` term referenced as part of describing the alex industry-euphemisms package; flagged at warning severity.
- `.wokeignore` — already lists `scout.personas.yaml` and `AGENTS.md` plus a comment block ("Self-referential: scout personas illustrate prohibited patterns in YAML comments"). Not honored by the action.
- `tools/generators/gen_action.py` `STATIC_FOOTER` — the generator that produces the consumer action. It has an "Inject canonical Open Paws paths into .wokeignore" step, but that step (a) only injects `.claude/rules/`, `CLAUDE.md`, `AGENTS.md`, not `scout.personas.yaml`, and (b) appends to `.wokeignore` in a way that the subsequent `woke --config /tmp/.woke.yaml ${{ inputs.paths }}` invocation doesn't actually read — empirically demonstrated by the failing run.
- `Open-Paws/no-animal-violence-action/action.yml` — propagated copy of the above. Same shape: `woke --exit-1-on-failure --config /tmp/.woke.yaml ${{ inputs.paths }}` with no `--exclude`, no `--ignore-files`, and no `ignore_files:` key inside the in-memory `/tmp/.woke.yaml` it generates. So whatever the consumer repo writes into `.wokeignore`, the action doesn't see it.

**Evidence:**

- Failing run: https://github.com/Open-Paws/no-animal-violence/actions/runs/24931266093 (PR #56)
- Same failure shape on PRs #55, #59, #61 (every open PR — verified by listing).

**Related issues / PRs:**

- #57 (CLOSED, verified) — "fix(gen_action): inject bundled .wokeignore allowlist for canonical Open Paws paths." Adds `.claude/rules/`, `CLAUDE.md`, `AGENTS.md` to a `.wokeignore` that the action's woke invocation doesn't actually consume; doesn't cover `scout.personas.yaml`. Symptom persists.
- #58 (OPEN, verified) — "feat(gen_action): add severity + fail-on-findings inputs, replace woke with scan.py invocation." Replaces woke with a `scan.py` invocation. If scan.py honors `.wokeignore` once it propagates downstream, that may fix this transitively — but `scan.py` does not yet exist in the repo (`ls /tmp/nav-check/scan.py` → No such file or directory) and the propagation chain to the consumer action hasn't completed. So this issue is still live regardless of #58's eventual landing.

**Source:** `source:human-report` — Sam, observed across 4 blocked PRs.

## Impact

All 4 open PRs blocked on the same false-positive CI failure. Every new PR will hit the same wall. This is a self-inflicted CI deadlock where the repo that defines the scanner cannot pass its own scanner because the scanner refuses to skip its own fixture catalog.

## Recommended fix direction (not for scout to decide)

Several plausible approaches — leaving option choice to triage / planner:

a. Make the action honor a repo-local `.wokeignore` (either by switching the consumer woke invocation off `--config /tmp/.woke.yaml` to a strategy that lets woke discover ignore paths from cwd, or by passing `--ignore-files` / equivalent explicitly).

b. Refactor `scout.personas.yaml` (and any other fixture catalog) to use abstract markers like `<TEST_PATTERN_KILL_TWO_BIRDS>` that get expanded to the real phrases at scanner test time only.

c. Move test fixtures under a dedicated path (e.g. `tests/fixtures/`) and have the action skip that path by default or via input.

d. Add an `allowlist-files` action input so consumer repos can declare per-repo exemptions cleanly.

e. Land `scan.py` (per #58) AND ensure scan.py respects `.wokeignore` when it propagates.

## Acceptance criteria

- [ ] PRs that don't touch flagged files pass `nav-check` cleanly on a representative branch from `Open-Paws/no-animal-violence`.
- [ ] The fixture content in `scout.personas.yaml` continues to exist (it's load-bearing — the persona catalog needs the phrases) OR is replaced with a mechanism that preserves the test-coverage intent.
- [ ] One representative re-run of `nav-check` on PR #55, #56, #59, or #61 shows green.
- [ ] No new false positives introduced for downstream consumers of `Open-Paws/no-animal-violence-action`.
- [ ] Solution is documented (either in `.wokeignore` comments, the action README, or `AGENTS.md`) so the next person hitting "why is the scanner skipping this file" finds the answer fast.

## Note on labels

Filer requested `auto:auto-fixable` + `sensitivity:public-ok`. Per `~/.claude/skills/scout-playbook/SKILL.md`, scout does not own `auto:*` or `sensitivity:*` — triage does. Honoring filer instruction here, but triage should re-validate.

STAGE 6 test review — approved.

Pulled `fix/65-nav-check-fixture-allowlist` at commit `a8c08d4`. Diff is test-only (`tools/generators/tests/test_generators.py` +283/-21, `gen_action.py` untouched per `git show --stat`). Locked-tests rule is uncompromised going into STAGE 7.

## RED verified locally

```
tools/generators/tests/test_generators.py::test_build_woke_yaml_emits_ignore_files_as_list_of_strings   FAILED
tools/generators/tests/test_generators.py::test_build_woke_yaml_ignore_files_contains_canonical_paths   FAILED
tools/generators/tests/test_generators.py::test_gen_action_woke_config_has_ignore_files                 FAILED
tools/generators/tests/test_generators.py::test_gen_action_woke_config_includes_scout_personas_yaml     FAILED
tools/generators/tests/test_generators.py::test_gen_action_woke_ignore_patterns_match_fixture_files     FAILED
tools/generators/tests/test_generators.py::test_gen_action_static_footer_drops_wokeignore_injection_step FAILED
```

All 6 new tests fail against unmodified `gen_action.py`. The 3 remaining failures (`test_vscode_word_boundary_emission`, `test_eslint_word_boundary`, `test_danger_word_boundary`) are pre-existing `Cannot find module 'js-yaml'` env failures — verified untouched by this commit and disclosed up-front by the test-writer.

## Plan v2 acceptance-criteria coverage

| ACMap item | Test |
|---|---|
| (1) Inline `ignore_files` list in woke YAML | `test_build_woke_yaml_emits_ignore_files_as_list_of_strings` (schema), `test_build_woke_yaml_ignore_files_contains_canonical_paths` (contents), `test_gen_action_woke_config_has_ignore_files` (end-to-end heredoc parse) |
| (2) `scout.personas.yaml` content preserved | absence-of-change (plan v2 explicit) |
| (3) Runtime nav-check green | STAGE 8 runtime verification (out of scope for unit tests) |
| (4) No new false positives | `test_gen_action_woke_ignore_patterns_match_fixture_files` `must_not_match` list (`src/main.py`, `README.md`, `docs/index.md`, `tools/generators/gen_action.py`) |
| (5) Documented | PR review (plan v2 explicit: not a runtime test) |
| Plan v2 cleanup | `test_gen_action_static_footer_drops_wokeignore_injection_step` |
| #65 regression | `test_gen_action_woke_config_includes_scout_personas_yaml` |

Every plan v2 mapping line is covered. No `NOT_TESTABLE` skips.

## Mutation resistance

Three-question rubric per `~/.claude/rules/testing.md`:

- **Schema test** (`emits_ignore_files_as_list_of_strings`) — asserts `isinstance(ignore_files, list)` AND per-element `isinstance(entry, str)`. Mutating value to `".wokeignore"` (the v1-plan bug) fails on the list check; mutating to a list of non-strings fails on per-element check. Also guards `rules` key not dropped. Strong.
- **Contract test** (`ignore_files_contains_canonical_paths`) — exact-equality membership across all four canonical paths. Drift like `scout.personas.yml` (extension) or `.claude/rules` (missing trailing slash) trips the assertion. Strong.
- **End-to-end contract** (`woke_config_has_ignore_files`) — extracts the embedded `/tmp/.woke.yaml` HEREDOC body via the `cat > /tmp/.woke.yaml << 'RULES'` opener, dedents, re-parses as YAML. This is the runtime contract surface — what woke 0.19.0 actually reads. Strong.
- **Regression** (`includes_scout_personas_yaml`) — focused single-path assertion. Domain rule: this is the #65 specific bug. Strong.
- **Integration** (`ignore_patterns_match_fixture_files`) — feeds emitted patterns into `pathspec.GitWildMatchPattern` (same gitignore algorithm woke uses via go-git). Tests `must_match` at root + at depth (`subdir/scout.personas.yaml`) + dir trailing-slash (`.claude/rules/some-rule.md`), AND `must_not_match` for normal files. This is the strongest test in the wave: it catches the planner-flagged risk where the right strings appear in syntactically wrong form (`scout.personas.yaml` vs `**/scout.personas.yaml`). Strong.
- **Cleanup** (`drops_wokeignore_injection_step`) — three forbidden artifacts: step name, sentinel comment, `>> .wokeignore` mutation command. Any of them coming back trips the test. Outer YAML still parses (deletion mustn't break the action). Strong.

None of the tests are tautological. None are output-equals-output snapshots. All encode named domain rules (woke `[]string` schema, gitignore pattern semantics, plan v2 cleanup contract).

## Notes for STAGE 7 (implementer)

- `_build_woke_yaml` already accepts `indent: int = 8` default; tests call it with `indent=0` to YAML-parse the raw output. Keep the kwarg signature compatible.
- The integration test uses `pytest.importorskip("pathspec")` — `pathspec` is installed locally (`1.1.0`) and the schema/contract tests cover the structural contract independently if CI ever lacks it. Worth ensuring CI has `pathspec` installed so the integration test runs there.
- Plan v2 asks for the entire \"Inject canonical Open Paws paths into .wokeignore\" step deletion. The test asserts the action.yml outer schema still parses post-deletion, so don't accidentally remove the trailing newline that separates the install + run steps.

Stage label stays `stage:tests-approved`. STAGE 7 implementer dispatch unblocked.

## STAGE 8 (Verifier) — clean

**Branch:** `fix/65-nav-check-fixture-allowlist` @ `a80bfe5`
**PR:** #66

### Local verification (worktree, full suite)

- **Lint** (`ruff check .`) — `All checks passed!`
- **Test** (`pytest tools/test_check_consistency.py -v`, the canonical CI suite) — **20/20 passed in 0.54s**
- **Typecheck** — not configured in this repo (no mypy in CI; `ruff` lint covers Python static checks)
- **Full pytest discovery** (`pytest -v`, beyond the CI suite) — 42 passed, 3 pre-existing failures in `tools/generators/tests/test_generators.py` (`test_vscode_word_boundary_emission`, `test_eslint_word_boundary`, `test_danger_word_boundary`)

### Pre-existing failures — NOT a regression

All three failures share root cause: `Error: Cannot find module 'js-yaml'` when invoking `node tools/generators/gen_*.js`. Confirmed by checking out the same test file from `origin/main` and reproducing the identical 3 failures with the same `MODULE_NOT_FOUND` stack. Local Node `node_modules/` not installed; not introduced by this PR; not in the canonical CI suite (ci.yml runs `pytest tools/test_check_consistency.py` only).

### PR #66 CI rollup — all green

- Lint: SUCCESS
- Test: SUCCESS
- nav-check: SUCCESS
- Code Quality Gate (>= 70): SUCCESS
- CodeQL (actions / python / javascript-typescript): SUCCESS
- Auto-merge gate: SUCCESS
- CodeRabbit: SUCCESS

Mergeable: MERGEABLE.

Verifier classification: **GREEN**. Leaving `stage:verified`. Advancing to STAGE 9.

--- a/tools/generators/gen_action.py
+++ b/tools/generators/gen_action.py
@@ -52,14 +52,6 @@ runs:
 STATIC_FOOTER = """\
         RULES
 
-    - name: Inject canonical Open Paws paths into .wokeignore
-      shell: bash
-      run: |
-        if ! grep -qF "# no-animal-violence-action: canonical paths" .wokeignore 2>/dev/null; then
-          printf '\\n# no-animal-violence-action: canonical paths\\n' >> .wokeignore
-          printf '.claude/rules/\\nCLAUDE.md\\nAGENTS.md\\n' >> .wokeignore
-        fi
-
     - name: Run animal violence language scan
       shell: bash
       run: |
@@ -68,9 +60,30 @@ STATIC_FOOTER = """\
           ${{ inputs.paths }}
 """
 
+# Canonical Open Paws paths emitted as woke `ignore_files` patterns (gitignore
+# syntax). These files exist by design across every Open Paws repo and contain
+# domain terms (personas, rule descriptions) that the scanner would otherwise
+# flag on every PR. Inlining them here is plan v2 for #65 — the previous
+# .wokeignore mutation step never propagated to the deployed action.yml.
+CANONICAL_IGNORE_FILES = [
+    "**/*personas.yaml",
+    "scout.personas.yaml",
+    "AGENTS.md",
+    "CLAUDE.md",
+    ".claude/rules/",
+]
+
 
 def _build_woke_yaml(rules: list[Rule], indent: int = 8) -> str:
-    """Render rules in woke YAML format, indented for heredoc embedding."""
+    """Render rules in woke YAML format, indented for heredoc embedding.
+
+    Emits two top-level keys:
+      - `ignore_files`: list of gitignore-style patterns telling woke to skip
+        the canonical Open Paws files (personas fixtures, AGENTS.md, CLAUDE.md,
+        .claude/rules/) at scan time. Per woke pkg/config/config.go this maps
+        to `IgnoreFiles []string` — must be a list, never a single string.
+      - `rules`: the rule entries woke matches against.
+    """
     pad = " " * indent
     rules_list = []
     for rule in rules:
@@ -87,7 +100,10 @@ def _build_woke_yaml(rules: list[Rule], indent: int = 8) -> str:
         entry["reason"] = rule.reason
         rules_list.append(entry)
     dumped = yaml.safe_dump(
-        {"rules": rules_list},
+        {
+            "ignore_files": list(CANONICAL_IGNORE_FILES),
+            "rules": rules_list,
+        },
         default_flow_style=False,
         sort_keys=False,
         allow_unicode=True,

--- a/tools/generators/tests/test_generators.py
+++ b/tools/generators/tests/test_generators.py
@@ -4,6 +4,7 @@ import subprocess
 import sys
 from pathlib import Path
 
+import pytest
 import yaml
 
 FIXTURES = Path(__file__).parent / "fixtures"
@@ -275,15 +276,140 @@ def test_danger_word_boundary(tmp_path):
     assert r'"\\bcuriosity' not in output
 
 
-def test_gen_action_allowlist_injection(tmp_path):
-    """gen_action injects a .wokeignore block that allowlists canonical AI config paths.
+CANONICAL_IGNORE_PATHS = [
+    "scout.personas.yaml",
+    "AGENTS.md",
+    "CLAUDE.md",
+    ".claude/rules/",
+]
 
-    The STATIC_FOOTER must contain a guarded append block that writes
-    .claude/rules/, CLAUDE.md, and AGENTS.md to .wokeignore so woke does not
-    flag speciesist-language rule definitions in those files. The injection is
-    idempotent (guarded by grep -qF on the marker line).
 
-    This test is RED until STAGE 7 adds the injection block to STATIC_FOOTER.
+def _extract_woke_yaml_body(action_yml_text: str) -> dict:
+    """Parse the embedded /tmp/.woke.yaml HEREDOC body out of a generated action.yml.
+
+    The action.yml ships a composite step that writes /tmp/.woke.yaml via a
+    bash heredoc:
+
+        cat > /tmp/.woke.yaml << 'RULES'
+        <yaml body, 8-space indented>
+        RULES
+
+    This helper extracts the body between the heredoc opener and the RULES
+    sentinel, dedents the 8-space indent woke needs, and parses the result as
+    YAML so tests can assert against the actual schema woke will see at runtime.
+    Returns the parsed dict.
+    """
+    opener = "cat > /tmp/.woke.yaml << 'RULES'"
+    assert opener in action_yml_text, (
+        f"action.yml has no {opener!r} heredoc opener — generator structure changed"
+    )
+    after_opener = action_yml_text.split(opener, 1)[1]
+    # The closing sentinel "RULES" sits on its own line, indented to match the
+    # cat invocation. Match it as the first line that is exactly whitespace + RULES.
+    body_lines: list[str] = []
+    for line in after_opener.splitlines()[1:]:
+        if line.strip() == "RULES":
+            break
+        body_lines.append(line)
+    else:  # pragma: no cover - defensive
+        raise AssertionError("RULES heredoc terminator not found after opener")
+    # Dedent: the body is indented 8 spaces inside the composite step's run: |.
+    # Use a permissive dedent so tests survive whitespace re-jiggering as long
+    # as every body line shares a common leading indent.
+    indent = min(
+        (len(line) - len(line.lstrip(" ")) for line in body_lines if line.strip()),
+        default=0,
+    )
+    dedented = "\n".join(line[indent:] for line in body_lines)
+    parsed = yaml.safe_load(dedented)
+    assert isinstance(parsed, dict), (
+        f"woke YAML body did not parse as a mapping (got {type(parsed).__name__})"
+    )
+    return parsed
+
+
+def test_build_woke_yaml_emits_ignore_files_as_list_of_strings():
+    """_build_woke_yaml emits an `ignore_files` key whose value is a list of strings.
+
+    Per woke source (pkg/config/config.go), `IgnoreFiles []string` is a list of
+    inline gitignore-style PATTERNS — not a pointer to a .wokeignore file. The
+    schema is `[]string`. A single-string value would silently fail at runtime
+    (woke would treat it as one literal pattern), so this test asserts both
+    presence and the list-of-strings shape.
+
+    Mutation resistance:
+      - Mutate `ignore_files: [...]` → omit key → fails on key check.
+      - Mutate `ignore_files: [...]` → `ignore_files: ".wokeignore"` (the
+        v1-plan bug) → fails on `isinstance(value, list)`.
+      - Mutate any list element to non-string → fails on per-element type.
+    """
+    from gen_action import _build_woke_yaml
+
+    rules = load_rules(FIXTURES / "rules_mini.yaml")
+    rendered = _build_woke_yaml(rules, indent=0)
+    parsed = yaml.safe_load(rendered)
+
+    assert isinstance(parsed, dict), (
+        f"_build_woke_yaml output did not parse as a mapping "
+        f"(got {type(parsed).__name__})"
+    )
+    assert "ignore_files" in parsed, (
+        "_build_woke_yaml output missing top-level 'ignore_files' key — "
+        "woke will not skip canonical Open Paws paths"
+    )
+    ignore_files = parsed["ignore_files"]
+    assert isinstance(ignore_files, list), (
+        f"'ignore_files' must be a list (woke schema is []string), got "
+        f"{type(ignore_files).__name__}: {ignore_files!r}"
+    )
+    assert len(ignore_files) >= 1, "'ignore_files' must not be empty"
+    for entry in ignore_files:
+        assert isinstance(entry, str), (
+            f"every 'ignore_files' entry must be a string, got "
+            f"{type(entry).__name__}: {entry!r}"
+        )
+
+    # Rules key must still be present alongside ignore_files (not replaced).
+    assert "rules" in parsed, (
+        "_build_woke_yaml dropped the 'rules' key when adding ignore_files — "
+        "woke would have nothing to scan for"
+    )
+    assert isinstance(parsed["rules"], list) and len(parsed["rules"]) >= 1, (
+        "'rules' key must remain a non-empty list"
+    )
+
+
+def test_build_woke_yaml_ignore_files_contains_canonical_paths():
+    """_build_woke_yaml's `ignore_files` list contains every canonical path.
+
+    Mutation resistance: dropping any one of the canonical paths flips this
+    assertion. Substring matching is rejected — exact equality on the entry
+    string catches drift like `scout.personas.yml` (wrong extension) or
+    `.claude/rules` (missing trailing slash, which changes gitignore semantics).
+    """
+    from gen_action import _build_woke_yaml
+
+    rules = load_rules(FIXTURES / "rules_mini.yaml")
+    parsed = yaml.safe_load(_build_woke_yaml(rules, indent=0))
+    ignore_files = parsed["ignore_files"]
+
+    for canonical in CANONICAL_IGNORE_PATHS:
+        assert canonical in ignore_files, (
+            f"canonical Open Paws path {canonical!r} missing from "
+            f"_build_woke_yaml ignore_files list (got {ignore_files!r})"
+        )
+
+
+def test_gen_action_woke_config_has_ignore_files(tmp_path):
+    """Generated action.yml's embedded woke YAML carries ignore_files inline.
+
+    End-to-end: generate the full action.yml, extract the /tmp/.woke.yaml
+    HEREDOC body, parse it, and assert it has `ignore_files` shaped as a list
+    of strings containing the canonical paths. This is the contract test —
+    `_build_woke_yaml` is internal, but the embedded heredoc body is the
+    runtime contract with woke 0.19.0.
+
+    Mutation resistance via YAML parse, not substring match.
     """
     from gen_action import generate
 
@@ -292,24 +418,160 @@ def test_gen_action_allowlist_injection(tmp_path):
     generate(rules, output_path)
     content = output_path.read_text()
 
-    # Structural guard: injection must not break the YAML schema
-    data = yaml.safe_load(content)
-    assert "runs" in data, "action.yml top-level 'runs' key missing — YAML structure broken"
-    assert "steps" in data["runs"], "action.yml runs.steps missing — YAML structure broken"
-
-    # Idempotency marker must be present
-    marker = "# no-animal-violence-action: canonical paths"
-    assert marker in content, (
-        f"Allowlist injection marker {marker!r} not found in generated action.yml"
+    # Outer action.yml schema must still parse.
+    outer = yaml.safe_load(content)
+    assert "runs" in outer and "steps" in outer["runs"], (
+        "action.yml outer YAML structure broken"
     )
 
-    # Each canonical path must appear AFTER the marker (not before, not absent)
-    marker_idx = content.index(marker)
-    for path in (".claude/rules/", "CLAUDE.md", "AGENTS.md"):
-        path_idx = content.find(path, marker_idx)
-        assert path_idx != -1, (
-            f"Canonical path {path!r} not found after allowlist marker in action.yml"
+    woke_config = _extract_woke_yaml_body(content)
+    assert "ignore_files" in woke_config, (
+        "embedded /tmp/.woke.yaml heredoc body missing 'ignore_files' — "
+        "woke will not skip fixture files at runtime"
+    )
+    ignore_files = woke_config["ignore_files"]
+    assert isinstance(ignore_files, list), (
+        f"embedded woke ignore_files must be a list, got {type(ignore_files).__name__}"
+    )
+    for canonical in CANONICAL_IGNORE_PATHS:
+        assert canonical in ignore_files, (
+            f"canonical path {canonical!r} missing from embedded woke "
+            f"ignore_files list (got {ignore_files!r})"
         )
+
+
+def test_gen_action_woke_config_includes_scout_personas_yaml(tmp_path):
+    """Regression test for #65: scout.personas.yaml MUST be in ignore_files.
+
+    The bug in #65 is specifically that `scout.personas.yaml` (a fixture file
+    full of personas with phrases like 'guinea pig' on purpose, to exercise
+    the rules) was being scanned by woke and flagged on every PR. The fix is
+    that scout.personas.yaml ends up in the woke config's ignore_files list.
+
+    This test fails specifically if a future refactor drops scout.personas.yaml
+    from the canonical paths list — even if the other three paths remain.
+    """
+    from gen_action import generate
+
+    rules = load_rules(FIXTURES / "rules_mini.yaml")
+    output_path = tmp_path / "action.yml"
+    generate(rules, output_path)
+
+    woke_config = _extract_woke_yaml_body(output_path.read_text())
+    ignore_files = woke_config.get("ignore_files", [])
+    assert "scout.personas.yaml" in ignore_files, (
+        "scout.personas.yaml is the #65 regression case — it MUST appear in "
+        f"the woke config ignore_files list. Got: {ignore_files!r}"
+    )
+
+
+def test_gen_action_woke_ignore_patterns_match_fixture_files(tmp_path):
+    """Integration test: emitted ignore_files patterns actually skip fixture files.
+
+    Simulates woke's runtime matching by feeding the generator's ignore_files
+    list into a gitignore-pattern matcher (the same algorithm woke uses via
+    go-git/go-git's gitignore parser, but in Python via `pathspec`). Builds a
+    fixture repo with scout.personas.yaml plus a non-fixture file, and asserts
+    that the patterns skip the fixture and do NOT skip the regular file.
+
+    This catches the failure mode where ignore_files contains the right path
+    strings but in a syntax that doesn't actually match (e.g. missing leading
+    slash, wrong glob form, trailing-slash semantics drift).
+
+    Mutation resistance: drop scout.personas.yaml from canonical paths -> the
+    fixture file is no longer matched -> test fails. Replace bare filenames
+    with absolute-only patterns -> bare files at deeper paths stop matching
+    -> test fails on the nested case.
+    """
+    pathspec = pytest.importorskip("pathspec")  # noqa: F841
+
+    from gen_action import _build_woke_yaml
+
+    rules = load_rules(FIXTURES / "rules_mini.yaml")
+    parsed = yaml.safe_load(_build_woke_yaml(rules, indent=0))
+    ignore_patterns = parsed.get("ignore_files", [])
+    assert isinstance(ignore_patterns, list) and ignore_patterns, (
+        "ignore_files must be a non-empty list to test runtime matching"
+    )
+
+    import pathspec as _pathspec
+    spec = _pathspec.PathSpec.from_lines(
+        _pathspec.patterns.GitWildMatchPattern,
+        ignore_patterns,
+    )
+
+    # Files that MUST match (be skipped by woke at runtime).
+    must_match = [
+        "scout.personas.yaml",          # at root — primary #65 case
+        "AGENTS.md",
+        "CLAUDE.md",
+        ".claude/rules/some-rule.md",   # under canonical rules dir
+        "subdir/scout.personas.yaml",   # bare-filename gitignore matches at any depth
+    ]
+    for path in must_match:
+        assert spec.match_file(path), (
+            f"emitted ignore_files patterns {ignore_patterns!r} do not match "
+            f"{path!r} — woke would scan it at runtime and #65 reproduces"
+        )
+
+    # Files that MUST NOT match (regular code/docs that should still be scanned).
+    must_not_match = [
+        "src/main.py",
+        "README.md",
+        "docs/index.md",
+        "tools/generators/gen_action.py",
+    ]
+    for path in must_not_match:
+        assert not spec.match_file(path), (
+            f"emitted ignore_files patterns {ignore_patterns!r} unexpectedly "
+            f"match {path!r} — over-broad ignore would silence real findings"
+        )
+
+
+def test_gen_action_static_footer_drops_wokeignore_injection_step(tmp_path):
+    """Plan v2 drops the never-propagated `.wokeignore` injection step.
+
+    Path A inlines the canonical paths into the woke YAML config's
+    `ignore_files:` key. The shell step that mutated consumer `.wokeignore`
+    files (a) never propagated to the deployed action.yml, and (b) is
+    redundant under Path A. Plan v2 explicitly removes it.
+
+    This test asserts the artifacts of that step are gone from the generated
+    action.yml. It fails on the current codebase (the step is still there)
+    and goes green when STAGE 7 deletes the step from STATIC_FOOTER.
+
+    Mutation resistance: any of the three step artifacts (step name, sentinel
+    comment, append-to-.wokeignore command) coming back trips the test.
+    """
+    from gen_action import generate
+
+    rules = load_rules(FIXTURES / "rules_mini.yaml")
+    output_path = tmp_path / "action.yml"
+    generate(rules, output_path)
+    content = output_path.read_text()
+
+    # Outer schema still valid (the deletion mustn't break the action).
+    data = yaml.safe_load(content)
+    assert "runs" in data and "steps" in data["runs"], (
+        "action.yml outer YAML structure broken after STATIC_FOOTER edit"
+    )
+
+    forbidden_artifacts = [
+        # The composite-step name.
+        "Inject canonical Open Paws paths into .wokeignore",
+        # The idempotency sentinel comment line.
+        "# no-animal-violence-action: canonical paths",
+        # The mutation command — appending to consumer .wokeignore.
+        ">> .wokeignore",
+    ]
+    for artifact in forbidden_artifacts:
+        assert artifact not in content, (
+            f"STATIC_FOOTER still contains {artifact!r} — plan v2 removes the "
+            "wokeignore-injection step entirely (canonical paths now live in "
+            "the woke YAML ignore_files key, not in consumer .wokeignore)"
+        )
+
+
 
 
 def test_gen_action_woke_command_present(tmp_path):

